src/main/java/org/opentest4j/FileInfo.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2015-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.opentest4j;
+
+import java.io.Serializable;
+import java.nio.charset.Charset;
+import java.util.Arrays;
+
+/**
+ * {@code FileInfo} is a pair of a file path and the contents of the file.
+ * <p>
+ * {@code FileInfo} is expected to be used as {@code actual} and {@code expected} values in {@code AssertionFailedError}
+ * <p>
+ * The semantics of the {@code path} can be defined by the users of this class.
+ *
+ * @author Marc Philipp
+ * @author Reinhold Degenfellner
+ * @since 1.2.1
+ */
+public class FileInfo implements Serializable {
+
+	private static final long serialVersionUID = 1L;
+
+	private final String path;
+	private final byte[] contents;
+
+	/**
+	 * Constructs a {@code FileInfo} with a path to a file and the contents of the file.
+	 * <p>
+	 * Be cautious with large files (maybe limiting the size is a good idea)
+	 *
+	 * @param path the path to the file; must not be {@code null}
+	 * @param contents the contents of the file; must not be {@code null};
+	 */
+	public FileInfo(String path, byte[] contents) {
+		if (contents == null) {
+			throw new NullPointerException("contents must not be null");
+		}
+		validatePath(path);
+		this.contents = contents;
+		this.path = path;
+	}
+
+	private static void validatePath(String path) {
+		if (path == null || path.trim().isEmpty()) {
+			throw new NullPointerException("path must not be null or blank");
+		}
+	}
+
+	/**
+	 * Returns the path to the file.
+	 *
+	 * @return the path to the file; never {@code null}
+	 */
+	public String getPath() {
+		return path;
+	}
+
+	/**
+	 * Returns the contents of the file.
+	 *
+	 * @return the contents of the file; never {@code null}
+	 */
+	public byte[] getContents() {
+		return contents;
+	}
```

Comment:
Similarly, this should either copy the contents when returned to prevent users from changing the inner contents, or it should be of type `String`, a custom `ByteString`, or an unmodifiable `List<Byte>` view.

What do you think?

Reply:
see above comment.